File: Tyche.Android/bet/src/main/java/com/felipearpa/tyche/bet/application/BetUseCase.kt
package com.felipearpa.tyche.bet.application

import com.felipearpa.tyche.bet.domain.Bet
import com.felipearpa.tyche.bet.domain.PoolGamblerBetRepository
import javax.inject.Inject

class BetUseCase @Inject constructor(private val poolGamblerBetRepository: PoolGamblerBetRepository) {

    suspend fun execute(bet: Bet) =
        poolGamblerBetRepository.bet(bet = bet)
}

File: Tyche.iOS/DataBet/Sources/DataBet/Application/BetUseCase.swift
import Core

public class BetUseCase {
    private let poolGamblerBetRepository: PoolGamblerBetRepository
    
    public init(poolGamblerBetRepository: PoolGamblerBetRepository) {
        self.poolGamblerBetRepository = poolGamblerBetRepository
    }
    
    public func execute(bet: Bet) async -> Result<PoolGamblerBet, Error> {
        return await poolGamblerBetRepository.bet(bet: bet)
    }
}
